egs/hi_mia/v1/local/run_fst.py

```diff
@@ -0,0 +1,215 @@
+
+import numpy as np
+import argparse
+import sys
+import copy
+class edge:
+    def __init__(self, cur_state, to_state, isymbol, osymbol):
+        self.cur_state=int(cur_state)
+        self.to_state=int(to_state)
+        self.isymbol=int(isymbol)
+        self.osymbol=int(osymbol)
+    
+    def getToState(self):
+        return self.to_state
+
+class state:
+    def __init__(self):
+        self.oedge=list()
+        self.inputSymbol=list()
+    def addEdge(self,edge):
+        self.oedge.append(edge)
+        self.inputSymbol.append(int(edge.isymbol))
+    def hasOut(self,inputId):
+        for i in self.inputSymbol:
+            if i== inputId:
+                return True
+        return False
+    
+    def getOutput(self, inputId):
+        if inputId in self.inputSymbol:
+            pos = self.inputSymbol.index(inputId)
+        else:
+            return -1,-1
+        if pos == -1:
+            sys.exit(1)
+        return self.oedge[pos].to_state, self.oedge[pos].osymbol
+
+class token:
+    def __init__(self):
+        self.active=False
+        self.is_filler=True
+        self.score=0
+        self.num_keyword_frames=0
+        self.average_keyword_score = 0.0
+        self.keyword = 0
+        self.num_frames_of_current_state = 0
+        self.num_keyword_states = 0
+        self.max_score_of_current_state = 0.0
+        self.average_max_keyword_score = 0.0
+        self.average_max_keyword_score_before = 0.0
+    def transform(self,prev,olabel,is_self_arc,is_filler,score):
+        if(self.active==False or (self.active==True and self.score < prev.score+score)):
+            self.score = prev.score+ score
+            if(is_filler==False):
+                t=prev.num_keyword_frames
+                average_keyword_score = (score + prev.average_keyword_score * t) / (t + 1); 
+                num_keyword_frames = t + 1
+                if is_self_arc:
+                    self.num_frames_of_current_state = prev.num_frames_of_current_state + 1
+                    self.num_keyword_states = prev.num_keyword_states
+                    self.max_score_of_current_state = max(prev.max_score_of_current_state, score)
+                    self.average_max_keyword_score_before = prev.average_max_keyword_score_before
+                else:
+                    self.num_frames_of_current_state = 1
+                    self.num_keyword_states = prev.num_keyword_states + 1
+                    self.max_score_of_current_state = score
+                    self.average_max_keyword_score_before = prev.average_max_keyword_score
+                self.average_keyword_score=(self.max_score_of_current_state + 
+                    self.average_max_keyword_score_before * (self.num_keyword_states - 1)) / self.num_keyword_states
+                if olabel != 0:
+                    self.keyword=olabel
+            self.active=True
+            self.is_filler= is_filler
+                
+    def Reset():
```

Comment:
naming convention for class methods should be consistent with others, e.g. lower case for the first letter